File: scan-app/app/src/main/java/com/lock/main/presenter/MainPresenterImpl.java
package com.lock.main.presenter;

import com.lock.main.model.MainBean;
import com.lock.main.view.MainView;

import framework.base.BasePresenterImpl;
import framework.utils.SPUtils;

/**
 * by y on 2017/2/16
 */

public class MainPresenterImpl extends BasePresenterImpl<MainView> implements MainPresenter {

    public MainPresenterImpl(MainView view) {
        super(view);
    }

    @Override
    public void registerUser(String userName, String passWord) {
        MainBean userBean = new MainBean();
        userBean.setUserName(userName);
        userBean.setPassWord(passWord);
        SPUtils.writeUser(userBean);
        view.setUser(SPUtils.readeUser());

        if (SPUtils.isLogin()) {
            view.registerError();
        } else {
            view.registerSuccess();
        }

    }

}

File: mvplifecycler/src/main/java/com/android/mvp/lifecycle/mvp/MainPresenterImpl.kt
package com.android.mvp.lifecycle.mvp

import com.android.mvp.lifecycle.base.BasePresenterImpl

/**
 * @author y
 * @create 2019/3/31
 */
class MainPresenterImpl(view: MainView) : BasePresenterImpl<MainView>(view), MainPresenter {

    override fun netWork() {
        mView?.showToast()
    }

}
